### DISCUSSION TITLE
Randomized fingerprint

### DISCUSSION THREAD

**Discussion by UnknownElen** [Alex313031/thorium#244] [General]
![20231017_21h03m20s_grim](https://github.com/Alex313031/thorium/assets/112108695/dab67447-7681-4d2a-a4cd-085fc4ab4a2f)

It would be nice to have if there where more privacy focused stuff like Randomized fingerprint

**Comment by gz83** [comment]
Thanks for the suggestion.

@UnknownElen

**Comment by Insidevk18** [comment]
I leave here the patches that Brave uses to randomize the fingerprint:

https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
https://community.brave.com/t/fingerprinting-how-do-we-know-its-actually-working/134536

**Comment by github59173** [comment]
Cromite (https://github.com/uazo/cromite) method is the best so far I have found. I strongly disagree with the idea that a distributed fingerprint for everyone to use is the best method. That method requires a large user base of that browser which is not always the case. The best method is one that scrambles the fingerprint every time a request is sent, every time the tab is refreshed, every time you go to a new website, etc etc. Always scrambling the fingerprint is best so it can never be used as metric to track someone. For testing you should check Creepjs (https://abrahamjuliot.github.io/creepjs/).

**Reply by gz83** [reply]
Do you know which patches in cromite are related to random fingerprints? If you know it, please provide it to us, if not, then we will investigate.

**Reply by github59173** [reply]
Unfortunately, no I don't.

**Comment by Insidevk18** [comment]
I hope to be able to contribute with the little I know, if this is of help in good time.

These are browser extensions that block fingerprinting:

* CanvasBlocker (Only Fiefox)
   As far as I know there is an extension that allows you to run firefox extensions in chrome.
   https://github.com/kkapsner/CanvasBlocker
   https://canvasblocker.kkapsner.de/
   https://addons.mozilla.org/es/firefox/addon/canvasblocker/

* Canvas Fingerprint Defender Extensions (Chrome and Firefox)
  ->Canvas Fingerprint Defender
         https://mybrowseraddon.com/canvas-defender.html
         https://chrome.google.com/webstore/detail/canvas-fingerprint-defend/lanfdkkpgfjfdikkncbnojekcppdebfp
         https://addons.mozilla.org/en-US/firefox/addon/canvas-fingerprint-defender/

  ->Font Fingerprint Defender
        https://mybrowseraddon.com/font-defender.html
        https://chrome.google.com/webstore/detail/font-fingerprint-defender/fhkphphbadjkepgfljndicmgdlndmoke
        https://addons.mozilla.org/en-US/firefox/addon/font-fingerprint-defender/

  ->AudioContext Fingerprint Defender
       https://mybrowseraddon.com/audiocontext-defender.html
       https://chrome.google.com/webstore/detail/audiocontext-fingerprint/pcbjiidheaempljdefbdplebgdgpjcbe
       https://addons.mozilla.org/en-US/firefox/addon/audioctx-fingerprint-defender/

  ->WebGL Fingerprint Defender
       https://mybrowseraddon.com/webgl-defender.html
       https://chrome.google.com/webstore/detail/webgl-fingerprint-defende/olnbjpaejebpnokblkepbphhembdicik
       https://addons.mozilla.org/en-US/firefox/addon/webgl-fingerprint-defender/

* Canvas Blocker Fingerprint Protect (Chrome and Firefox)
     https://github.com/joue-quroi/canvas-fingerprint-blocker/
     https://add0n.com/canvas-fingerprint-blocker.html
     https://chrome.google.com/webstore/detail/canvas-blocker-fingerprin/nomnklagbgmgghhjidfhnoelnjfndfpd
     https://addons.mozilla.org/es/firefox/addon/canvas-blocker-no-fingerprint/
    
WITH THIS EXTENSION YOU CAN ER THE CODE OF THE EXTENSIONS:
* Extension Source Downloader
  https://mybrowseraddon.com/extension-source-downloader.html
  https://chrome.google.com/webstore/detail/extension-source-download/dlbdalfhhfecaekoakmanjflmdhmgpea

I hope this information is helpful :)
